### PR TITLE
 Make the root_link the father of the base_link instead of the other way around

### PR DESCRIPTION
**Pull request by traversaro**
This permits to avoid the Gazebo bug described in #140 . 
Furthermore, this will make sure that even after fixed joint lumping the link in Classic Gazebo will be called "root_link".

For an extensive rationale, see https://github.com/robotology/icub-model-generator/issues/140#issuecomment-700653608 .

This PR requires https://github.com/robotology/icub-model-generator/pull/141 to ensure that even if merge the changes will not reach directly the `master` branch.

### CODE REVIEW

**Comment by traversaro** [comment]
@Nicogene probably it could make sense to to the same change in the iCub 3 model.

**Comment by Nicogene** [comment]
Do you mean https://github.com/robotology/icub-model-generator/pull/142/commits/9e94025ab70cf8b1569f6edfcdf8e8df63146513 ? 

Or also the changes in the yaml.in?

**Comment by traversaro** [comment]
I mean just the change in the yaml.in, the commit you link is part of another PR (https://github.com/robotology/icub-model-generator/pull/141).

File: simmechanics/data/icub2_5/ICUB_2-5_BB_simmechanics_options.yaml.in
Comment (prashanthr05) [inline simmechanics/data/icub2_5/ICUB_2-5_BB_simmechanics_options.yaml.in:951]: Possibly we should document this decision somewhere more visible than in the issues?

Comment (traversaro) [reply simmechanics/data/icub2_5/ICUB_2-5_BB_simmechanics_options.yaml.in:951]: I plan to add a CHANGELOG.md in icub-models as done for all the other repos.

**Review by prashanthr05 — APPROVED** [review]
Ok for me! Maybe we can wait for another reviewer's (mainly @Nicogene being one of the active contributors for this repo) approval before proceeding for the merge.

**Review by Nicogene — APPROVED** [review]
Ok for me too, I tried it successfully with the icub 3 upper body model 👍

**Comment by prashanthr05** [comment]
@traversaro shall we then delete the branch?

**Comment by traversaro** [comment]
> @traversaro shall we then delete the branch?

Done!

**Pull request merged**
